ts/component/text-input.ts

```diff
@@ -20,14 +20,22 @@ export class TextInput {
     label.innerHTML = `${title}:`;
     const submitIcon = makeIcon("check");
     const textInputClickout = makeClickout();
+    textInputClickout.classList.add("absolute", "z-10");
     textInputClickout.classList.toggle("hidden");
 
     this.container.className =
       "font-inter font-medium text-12 text-black px-2 py-1 rounded-md ring-2 ring-offset-1 ring-red inline-block flex-auto items-center space-x-2";
     this.textField.className =
-      "ring-0 border-0 outline-0 rounded-sm relative px-1 min-w-[150px]";
+      "ring-0 border-0 outline-0 rounded-sm relative px-1 min-w-[150px] z-40";
     this.textField.setAttribute("size", "10");
-    submitIcon.classList.add("text-black", "hover:text-green", "relative");
+    submitIcon.classList.add(
+      "text-black",
+      "hover:text-green",
+      "relative",
+      "z-40"
+    );
+    label.classList.add("relative", "z-40");
```

Comment:
This is much better, but there is still an issue if you click part of the component that is not covered by these individual elements. This could easily happen by accident, like if you click just above or below the submit button, but still within the component. Nothing should happen if you click there, but currently the clickout is over-top of it, so it cancels the input. The solution we discussed previously of adding an additional container around the component and clickout would fix this and also reduce the number of places that z-index needs to be set:

```
<div> <!-- new outer container -->
  <div class="... z-40> <!-- existing container -->
    <span .../>
    <input .../>
    <i .../>
  </div>
  <button ... class="... z-10" .../> <!-- clickout -->
</div>
```